beaker/src/output_manager.rs

```diff
@@ -126,6 +141,41 @@ impl<'a> OutputManager<'a> {
         Ok(output_path)
     }
 
+    /// Generate main output path and track it as produced
+    pub fn generate_and_track_main_output(
```

Comment:
I think this should just be a track=true argument to generate main output, and we don't need 2 functions. We should be tracking outputs anyway so let's make that easier to happen by default